server/src/main/java/org/jboss/as/server/controller/resources/ModuleLoadingResourceDefinition.java

```diff
@@ -55,7 +56,8 @@
  */
 public class ModuleLoadingResourceDefinition extends SimpleResourceDefinition {
 
-    private static final AttributeDefinition MODULE_NAME = new SimpleAttributeDefinitionBuilder(MODULE, ModelType.STRING).build();
+    private static final AttributeDefinition MODULE_NAME = new SimpleAttributeDefinitionBuilder(MODULE,
+            ModelType.STRING).setCorrector(ModuleIdentifierUtil.MODULE_NAME_CORRECTOR).build();
```

Comment:
With this in place, we can now remove the canonicalization at:

https://github.com/wildfly/wildfly-core/blob/de60453b8a5c0213c6781093e1b1b1460b8f08bd/server/src/main/java/org/jboss/as/server/controller/resources/ModuleLoadingResourceDefinition.java#L210